harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -61,10 +66,49 @@ def unique_vm_name(unique_name):
     return f"vm-{unique_name}"
 
 
+@pytest.fixture(scope="class")
+def stopped_vm(api_client, ssh_keypair, wait_timeout, image, unique_vm_name):
+    unique_vm_name = f"stopped-{unique_vm_name}"
+    cpu, mem = 1, 2
+    pub_key, pri_key = ssh_keypair
+    vm_spec = api_client.vms.Spec(cpu, mem)
+    vm_spec.add_image("disk-0", image['id'])
+    vm_spec.run_strategy = "Halted"
+
+    userdata = yaml.safe_load(vm_spec.user_data)
+    userdata['ssh_authorized_keys'] = [pub_key]
+    vm_spec.user_data = yaml.dump(userdata)
+
+    code, data = api_client.vms.create(unique_vm_name, vm_spec)
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get(unique_vm_name)
+        if data.get('status'):
+            break
+        sleep(1)
```

Comment:
Do we want to implement the `sleep_timeout` fixture for this?

Reply:
Consider some cases:
1. Clone VM having large Volume
2. Get latest resource of created stopped VM
3. Download Image
4. Check Backup status

For 1st case, we would expect that it will few minutes to complete, then we can snooze more seconds to reduce API calls, same as case 3 and 4.
But for 2nd case, Create a stopped VM would be completed in a short time, and we could call API more frequently to make sure the VM is created.

So as I comment in other PR, we should not use fixture to make the seconds fixed, but let it be able to adjust on demend.

Reply:
Would that adjustment be that we have something like:
```python
sleep(sleep_timeout + 3)
```
If we wanted to sleep for 3 more seconds than the "fixed" base?
Or 
```python
sleep(sleep_timeout - 1)
```
If we wanted to sleep for 1 second less than the "fixed" base?

I'm thinking I could update my open PR to handle that if that's something we wanted to do.

I suppose it does get a little confusing with something like:
```python
def test_abc(request, api_client, wait_timeout, sleep_timeout):
    ...
    sleep(sleep_timout +2)
    ...
``` 
as reading through `test_abc()` we'd have to navigate over to the `config.yml` in order to find what the "base"/"fixed" sleep timeout was, then remember that and do additional math while reading through the lines in the function of `test_abc()`.
Whereas, when we do `sleep(1)` it's just very clear and apparent what the amount of sleep is before the next iteration.
So I like that as well :smile: 

Maybe if the need surfaced we could have an exponential backoff sleep, maybe some utility method that then returns a number based around the last iteration...perhaps?

Reply:
We should notice that config options could be updated by the test executor, so when we use:

```
sleep(sleep_timeout - 3)
```

The `sleep_timeout` might be negative if config option be updated to 2 or less.